engine/src/main/java/org/hibernate/validator/internal/engine/cascading/LegacyOptionalValueExtractor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.cascading;
+
+import java.util.Optional;
+
+import org.hibernate.validator.spi.cascading.ValueExtractor;
+
+/**
+ * Extractor for supporting the legacy construct {@code @Valid Optional<Foo>}.
+ * @author Gunnar Morling
+ *
```

Comment:
Sorry for the nitpicking but it's the only thing you let me: looks like you got the new line wrong in the javadoc comment.